src/desul/utility.hpp

```diff
@@ -0,0 +1,46 @@
+#if !defined(DESUL_UTILITY_HPP)
+#define DESUL_UTILITY_HPP
+
+#include <cstddef>
+#include <type_traits>
+
+namespace desul {
+   template <class T, T... Ints>
+   struct integer_sequence {
```

Comment:
How much of this do we need to recreate? We can probably just bring most into desul via using.

Reply:
I think it depends on whether or not we want to require the --expt-relaxed-constexpr compiler flag with nvcc and if there is an equivalent to that for rocm/hip compilers.

Reply:
But anything that does not have explicitly defined functions (like most type traits), I'm definitely supportive of bringing them in with using statements.

Reply:
I see it in the comments, but if you need integer_sequence stuff, take that wholesale from camp or somewhere.  It's a serious pain to get `make_integer_sequence` to perform well portably across compilers, and to cover some of the corner cases, no reason to go through that pain again.

Reply:
I'm going to leave it out for now and bring it in from camp in a future merge request.